instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -890,7 +890,9 @@ def instrument_client(
                 that is called right before the span ends
         """
 
-        if getattr(client, "_is_instrumented_by_opentelemetry", False):
+        if getattr(
+            client._transport, "_is_instrumented_by_opentelemetry", False
+        ):
             _logger.warning(
```

Comment:
We shouldn't log a warning if someone instruments a new client with the same transport. So I think we need to keep the attribute and warning on the client, but silently do nothing if the transport is already instrumented.